components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -1974,6 +1974,7 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         // Note: this operation is per-user only, hence needs no resource guard
         const user = this.checkAndBlockUser("createTeam");
         const team = await this.teamDB.createTeam(user.id, name);
+        const invite = await this.getGenericInvite(ctx, team.id);
```

Comment:
Retrieval of the `genericInvite` seems to also reset the invite link. Is this potentially a problem if the invite has already been sent? I'm not intimately familiar with the product/code behaviour in this area so seeking clarification.

https://github.com/gitpod-io/gitpod/blob/db8d56c49157d6f242778a4b69fbc5b8fc5f0d4b/components/server/src/workspace/gitpod-server-impl.ts#L2059

Reply:
hi @easyCZ , thanks for having a look! i went back into the preview environment and checked, the invite URL attached to the team created call here does not change unless the reset invite button is clicked below (at which point any previously sent invites lose their validity). thus, using `this.getGenericInvite()` is fine here imo as:

- no invites to a team can be sent prior to its creation
- the behaviour after this call is consistent with what's happening now as the invite URL remains the same unless it's manually reset

<img width="1332" alt="Screenshot 2022-03-28 at 10 54 34" src="https://user-images.githubusercontent.com/59581255/160362609-9cef16e3-e082-4081-b208-e4e6b5c0e3f0.png">